internal/driver/custommetadata.go

```diff
@@ -99,21 +94,20 @@ func (onvifClient *OnvifClient) getSpecificCustomMetadata(device contract.Device
 
 // deleteCustomMetadata will delete specified entries in custom metadata
 func (onvifClient *OnvifClient) deleteCustomMetadata(device contract.Device, data []byte) (contract.Device, errors.EdgeX) {
-	device = onvifClient.initCustomMetadata(device)
+	onvifClient.initCustomMetadata(&device)
 
 	var input []string
 	err := json.Unmarshal(data, &input)
 	if err != nil {
 		return device, errors.NewCommonEdgeX(errors.KindServerError, "failed to unmarshal the json request body", err)
 	}
 	if len(input) == 0 {
-		return device, errors.NewCommonEdgeX(errors.KindContractInvalid, "no data in query body", err)
+		return device, errors.NewCommonEdgeX(errors.KindContractInvalid, "no data in query parameter", err)
```

Comment:
I believe the Delete call is a `PUT` right? then it should be `request body`.

```suggestion
		return device, errors.NewCommonEdgeX(errors.KindContractInvalid, "no data in request body", err)
```